### PR TITLE
Chewing Gum

### PR DESCRIPTION
**Pull request by Shadowmech88**
gum_pack gum_wrapped gum gum_chewed gum_used_up
Adds chewing gum, which can be equipped to the mask slot in order to be chewed. While chewing, the reagents inside the gum will be slowly delivered to you, similar to cigarettes. Also similar to cigarettes, they can be dipped in reagents, up to a maximum of 15u. All the reagents are delivered in roughly a minute or so, after which the gum "loses its flavor" and can be discarded.
The gum will gradually lose color as it is drained of reagents, as well.
Stepping on a chewed piece of gum will cover your shoes/feet in gum.

Also adds explosive chewing gum, purchasable from a Syndicate uplink for 8 TC, twice the price of C4. Explosive chewing gum will detonate five seconds after you begin chewing it, and can be stuck to walls and objects like C4. Unlike C4, it is applied immediately, and cannot be directly placed on mobs. However, if a mob has stepped in it or is chewing it when it explodes, it will gib them.

:cl:
 * rscadd: Added chewing gum, which can be dipped in reagents like cigarettes and then equipped to the mask slot to have them slowly delivered to you.
 * rscadd: Added explosive chewing gum to Syndicate uplinks for 8 TC. It explodes five seconds after you begin chewing, and can be applied to walls and objects, but not mobs.

### CODE REVIEW

**Comment by Shadowmech88** [comment]
I'm not sure about where/how to make regular chewing gum accessible. Vending machines?

**Comment by ghost** [comment]
Nothing wrong with making them available in vending machines. What about adding it to the chef's candy maker too?

**Comment by Talvan** [comment]
Is it available via the chef's candy maker?

**Comment by Shadowmech88** [comment]
The chef's candy maker is garbage and I'd rather just make a proper recipe for it if I was to give it to the chef.

**Comment by ghost** [comment]
Smacking gum is fun, will we be able to blow bubbles in the future?

**Comment by Shadowmech88** [comment]
It shouldn't be too difficult to add a bubble gum variant later that lets you blow bubbles via an action button.

**Comment by ghost** [comment]
Aw yiss

**Comment by ghost** [comment]
you should be able to stick it into people's hair to make it perma messy until they shave it

**Comment by Intigracy** [comment]
No syringe transferring?

**Comment by Shadowmech88** [comment]
Syringe transferring is handled by syringes, isn't it?
Do I have to manually whitelist gum for syringes?

**Comment by SonixApache** [comment]
Getmore can sell gum, dan can sell gum with toxic waste in it.

**Comment by salamandooo** [comment]
me pops her gum

**Comment by Shadowmech88** [comment]
* Getmore vendors now carry 10 packs of chewing gum, sold for 10 credits each. Each pack of chewing gum contains 10 sticks of gum.
* Packs of chewing gum and sticks of chewing gum can now be injected with syringes.

**Comment by Blithering** [comment]
Looks neat, but five seconds isn't long for the fuse timer

**Comment by Shadowmech88** [comment]
I wanted it just long enough for you to have time to take it out of your mouth and stick it to something.
It's supposed to be an explosion that you can deploy quickly, not a time bomb.

Very useful for breaching walls and doors for a quick entry/getaway.

**Comment by Orangerush** [comment]
I'll be fully on board once I can repair damage by covering it up with bubblegum like In a cartoon

**Comment by Intigracy** [comment]
Agent Cody Banks

File: code/game/objects/items/gum.dm [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+			target.ex_act(1)
+		if(isobj(target))
+			if(target)
+				qdel(target)
+				target = null
+	qdel(src)
+
+/obj/item/gum/proc/chew(mob/user)
+	if(!chewed)
+		flags &= ~NOREACT //Allow reagents to react after being chewed
```
Comment (Intigracy) [inline code/game/objects/items/gum.dm:119]: Nice touch

Comment (Shadowmech88) [reply code/game/objects/items/gum.dm:119]: I just copied cigarette code.

**Pull request merged**